### ISSUE TITLE
train!

### ISSUE THREAD

**Issue by lannisite110** [denizyuret/Knet.jl#398]
I cannot understand well about the usage of train!(). Can you explain the "model"? I don't konw what it means.  what is a callable object?
`
help?> train!
search: train! Train train_up trailing_ones trailing_zeros AbstractString

  train!(model, data; loss, optimizer, callback, o...)

  Train a model with given data.

    •    model: A callable object. model(x; o...) should return a
        prediction. params(model) will automatically iterate over model
        parameters.

    •    data: An iterator. for (x,y) in data should iterate over
        input-output pairs.

    •    loss=nll: A loss function, called with J = @diff loss(model,x,y;
        o...).

    •    optimizer=Adam(): An optimizer object that will be copied for each
        parameter and used by [update!](@ref).

    •    callback: To facilitate reporting and termination, a callback
        function is called before every update with callback(J). Training
        continues if the return value is true, terminates if it is false.
        The default callback runs until training loss convergence.

    •    Other keyword arguments (o...) will be passed to loss and possibly
        by loss to model.
`

**Comment by denizyuret** [comment]
Please look at the tutorial for examples.
